[stable/kong] Added support for Kong datastore password config through secrets

tmestdagh opened the pull request:
It is now also possible to use a map for `kong.env` params, effectively allowing use of a secret instead of the password itself to configure the datastore backend.

shashiranjan84 reviewed (commented):
Overall lgtm, can we a test for it?

tmestdagh committed:
[stable/kong] Added support for Kong datastore password config through
secrets

Bumped version.

Updated README with some example usage of configuring `kong.env` parameters through a Secret.

Signed-off-by: Tom Mestdagh <mestdagh.tom@gmail.com>
